Makefile

```diff
@@ -114,21 +118,28 @@ e2e.test:
 test-e2e: e2e.test
 	./e2e.test -test.timeout 180m -test.v -ginkgo.v
 
-test-go: generate
-	go build -tags containers_image_openpgp ./...
+test-go: generate build-all validate-go lint-go unit-test-go coverage-go
 
+.PHONY: validate-go
+validate-go:
 	gofmt -s -w cmd hack pkg test
 	go run ./vendor/golang.org/x/tools/cmd/goimports -w -local=github.com/Azure/ARO-RP cmd hack pkg test
 	go run ./hack/validate-imports cmd hack pkg test
 	go run ./hack/licenses
 	@[ -z "$$(ls pkg/util/*.go 2>/dev/null)" ] || (echo error: go files are not allowed in pkg/util, use a subpackage; exit 1)
 	@[ -z "$$(find -name "*:*")" ] || (echo error: filenames with colons are not allowed on Windows, please rename; exit 1)
 	@sha256sum --quiet -c .sha256sum || (echo error: client library is stale, please run make client; exit 1)
-	go test -tags e2e -run ^$$ ./test/e2e/...
-
 	go vet ./...
-	set -o pipefail && go test -v ./... -coverprofile cover.out | tee uts.txt
 
+.PHONY: unit-test-go
+unit-test-go:
```

Comment:
Is there any benefit of introducing all these new make targers? Our `Makefile` already has too many targets. I would like to keep it useful for humans.

Reply:
I firmly believe there is. Instead of one monolithic `test-go`, there are now granular steps with the individual steps. This allows for CI to call these individual steps with the same code as developer uses locally. 

I understand the split add more code to already big makefile, which can be perceived as complicating the code. I would like to clarify, my intention with this PR is the exact opposite. Simplify the flow for the CI and unify the calls between localhost and CI.

I have achieved it by splitting the original `test-go`. To preserve original `test-go` functionality, to ensure some hidden call is not broken, I added all individual parts as dependency to the `test-go`.